config/config.go

```diff
@@ -828,6 +836,47 @@ func (c *KubernetesSDConfig) UnmarshalYAML(unmarshal func(interface{}) error) er
 	return nil
 }
 
+// GCESDConfig is the configuration for GCE based service discovery.
+type GCESDConfig struct {
+	// Project: The Google Cloud Project ID
+	Project string `yaml:"project"`
+
+	// Zone: The zone of the scrape targets.
+	// If you need to configure multiple zones use multiple gce_sd_configs
+	Zone string `yaml:"zone"`
+
+	// Filter: Can be used optionally to filter the instance list by other criteria.
+	// Syntax of this filter string is described here in the filter query parameter section:
+	// https://cloud.google.com/compute/docs/reference/latest/instances/list
+	Filter string `yaml:"filter,omitempty"`
```

Comment:
Yes, that makes regular expressions easier. See
http://www.robustperception.io/little-things-matter/

On Thu, Jun 30, 2016 at 2:31 PM, Ingo Gottwald <notifications@github.com>
wrote:

> In config/config.go
> <https://github.com/prometheus/prometheus/pull/1769#discussion_r69185745>:
>
> > @@ -828,6 +836,47 @@ func (c *KubernetesSDConfig) UnmarshalYAML(unmarshal func(interface{}) error) er
> >  	return nil
> >  }
> >
> > +// GCESDConfig is the configuration for GCE based service discovery.
> > +type GCESDConfig struct {
> > +	// Project: The Google Cloud Project ID
> > +	Project string `yaml:"project"`
> > +
> > +	// Zone: The zone of the scrape targets.
> > +	// If you need to configure multiple zones use multiple gce_sd_configs
> > +	Zone string `yaml:"zone"`
> > +
> > +	// Filter: Can be used optionally to filter the instance list by other criteria.
> > +	// Syntax of this filter string is described here in the filter query parameter section:
> > +	// https://cloud.google.com/compute/docs/reference/latest/instances/list
> > +	Filter string `yaml:"filter,omitempty"`
>
> So just a joined string. Just to be sure, "," at the beginning and at the
> end?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/prometheus/prometheus/pull/1769/files/aaca980fcffb0da8ac9d3711ab33e4abfb536610..aaf0f438d9011e3d02fff333c738551ab9d491d0#r69185745>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAANaJIJ3K4TQiHKRbhcURmCKY-LGiBMks5qRAtsgaJpZM4JAGOO>
> .
>